src/app/service/cityio.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http'
+import { interval } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CityioService {
+
+  baseUrl = "https://cityio.media.mit.edu/api/table/";
+  tableName = "virtual_table";
+  url = `${this.baseUrl}${this.tableName}`;
+
+  table_data: any = null; // can be accessed by other components, this will always be updated
+
+  update = interval(1000); // this is just rsjs way of a interval
+
+  constructor(private http: HttpClient) { 
+    this.update.subscribe((n)=>{
+      this.http.get(this.url).subscribe((data)=> this.table_data = data);
+    }) 
```

Comment:
digging into more documentation there is `flatmap` to chain observables

``` typescript
this.update.flapmap((n: Number)=>{
  return this.http.get(this.url);
}).subscribe((data: any) => {
  this.table_data = data;
});
```
I won't change the code, since it the above code works and actually straight forward once
you see too observables nested.